### PR TITLE
ModelScript methods

### PR DESCRIPTION
**Pull request by JaXt0r**
I implemented the full data export for ModelScript.

(Tedious work. 😆 )

Some hints:

* I tried to combine elements wherever a full property set could be exported within one method. e.g.
```c++
PXC_API void pxMdsGetAnimation_EventPfxStop(PxModelScript* mds, uint32_t animIndex, uint32_t pfxStopIndex,
	int32_t* frame,
	int32_t* index);
```

* As there were vectors within vectors, (e.g. ```model_script.animations[].events[]_```) I added an underscore to highlight that it's a sublist of a list. e.g.
```c++
int32_t pxMdsGetAnimation_EventTagFrame(PxModelScript* mds, uint32_t animIndex, uint32_t eventTagIndex);
```

* C# endpoint is developed and works already.

Feel free to suggest improvements

### CODE REVIEW

File: .editorconfig
Comment (lmichaelis) [inline .editorconfig:1]: If you feel like an EditorConfig is necessary, please add it in a separate PR.

File: include/phoenix/cffi/ModelScript.h [hunk 1]
```diff
@@ -14,7 +14,168 @@ typedef phoenix::model_script PxModelScript;
...
+typedef enum {
+	fist,       ///< The player fights with his fists.
+	one_handed, ///< The player wields a one-handed weapon.
+	two_handed, ///< The player wields a two-handed weapon.
+	bow,        ///< The player wields a bow.
+	crossbow,   ///< The player wields a crossbow.
+	magic,      ///< The player casts a magic spell.
+	none,       ///< The player is not in a fighting stance.
+	invalid,    ///< A fight mode which acts as an `unset` marker. Added for OpenGothic compatibility.
+} PxEventFightMode;
```
Comment (lmichaelis) [inline include/phoenix/cffi/ModelScript.h]: Enum members must be prefixed with `Px` too. Ideally the naming convention should be for example:

```c
typedef enum {
   PxAnimationDirectionForward = 0,
   PxAnimationDirectionBackward = 1,
} PxAnimationDirection;
```

Also, all enum members should have an assigned value. I forgot to add that on the phoenix side but I'll do that later. For now, just number them sequentially from 0, please.

**Review by lmichaelis — CHANGES_REQUESTED** [review]
Thanks! I'm unsure about the `pxMdsGetAnimation_` scheme but it's fine for now. There are some formatting issues but I'll fix them myself later :) Also, you need to cast the return value of `.size()`-calls to `uint32_t` explicitly. Otherwise the compilers will complain.

**Comment by JaXt0r** [comment]
Thanks for the feedback, @lmichaelis . I'll take care.

One question about the linter: Is there a way clang-lint can tell us what it complains about and not just where? I want to comply with it's demands, but never know what it is about as I also don't use it locally. 😉

**Comment by lmichaelis** [comment]
> Thanks for the feedback, @lmichaelis . I'll take care.
> 
> One question about the linter: Is there a way clang-lint can tell us what it complains about and not just where? I want to comply with it's demands, but never know what it is about as I also don't use it locally. wink

I can look into it. Most likely it'll just be a diff showing what needs to be changed. Hmmm maybe I'll make it automatically add a commit in pull requests ..

**Comment by JaXt0r** [comment]
I took care of your requested changed. And tested it.

The only thing is again: clang-lint. :-) 
If you can at least share with me the lint settings, then I can take care of it myself.

One idea: Do you require CRLF? (As my 2 files are LF).

File: include/phoenix/cffi/ModelScript.h [hunk 2]
```diff
@@ -14,7 +14,168 @@ typedef phoenix::model_script PxModelScript;
...
+	PxAnimationFlagsRotate = 2,
+	PxAnimationFlagsQueue = 4,
+	PxAnimationFlagsFly = 8,
+	PxAnimationFlagsIdle = 16  
+} PxAnimationFlags;
+
+typedef enum : uint8_t {
+	PxAnimationDirectionForward = 0,  ///< The animation samples are played from first to last.
+	PxAnimationDirectionBackward = 1, ///< The animation samples are played from last to first.
+} PxAnimationDirection;
```
Comment (lmichaelis) [inline include/phoenix/cffi/ModelScript.h]: C does not support typed enums. Remove `: uint8_t`.

**Review by lmichaelis — CHANGES_REQUESTED** [review]
> If you can at least share with me the lint settings, then I can take care of it myself.

The `.clang-format`-file contains the style definitions.

> One idea: Do you require CRLF? (As my 2 files are LF).

Everything in Git should always be LF only (and that's the case here).

**Comment by JaXt0r** [comment]
Hi  @lmichaelis 

I fixed all your code requests.

I also executed clang-lint locally and fixed a lot of formatting.
Now it fails on one additional method where I can't see why.

One idea: The method itself has an opening bracket not directly at 4-spaces/tab. Maybe this is the reason?

This is by the way the best bet I found on how to get more details about clang-format. But it didn't help me unfortunately:
https://discourse.llvm.org/t/make-clang-format-give-detailed-error-when-there-is-a-format-problem/5124/4

Please help!

**Pull request merged**